projects/graph/graph.py

```diff
@@ -7,42 +7,112 @@ class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
         self.vertices = {}
+        self.queue = Queue()
+        self.stack = Stack()
     def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
-        pass  # TODO
+        self.vertices[vertex] = set()
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
-        pass  # TODO
+        if v1 not in self.vertices:
+            self.add_vertex(v1)
+        else:
+            self.vertices[v1].add(v2)
+
```

Comment:
How would you go about refactoring `add_edges()` to work with an `undirected` graph?